#83 transaction dev-30 added (#129)
IHE/sdpi-fhir: SDPi_Supplement/asciidoc/volume2/dev-30/tf2-dev-30.adoc

:var_transaction_id: DEV-30

:var_label_dev_30_message_getmdib: GetMdib
:var_label_dev_30_message_getmdibresponse: GetMdibResponse

[#vol2_clause_dev_30]
=== Retrieve BICEPS Content [{var_transaction_id}]

==== Scope

This transaction is used to retrieve the MDIB of a <<actor_somds_provider>> a <<actor_somds_consumer>> is interested in.

==== Actor Roles

.Actor Roles {var_transaction_id}
[cols="1,2"]
|===
|Actor |Roles

|<<actor_somds_consumer>>

> |Initiates communication by sending <<vol2_clause_dev_30_message_getmdib, {var_label_dev_30_message_getmdib}>> messages to <<actor_somds_provider>>s. Then listens for <<vol2_clause_dev_30_message_getmdibresponse, {var_label_dev_30_message_getmdibresponse}>> messages to retrieve the <<actor_somds_provider>>'s MDIB.

Reviewer: ```suggestion
|Initiates communication by sending a <<vol2_clause_dev_30_message_getmdib, {var_label_dev_30_message_getmdib}>> message to a <<actor_somds_provider>>. Then listens for a <<vol2_clause_dev_30_message_getmdibresponse, {var_label_dev_30_message_getmdibresponse}>> message to retrieve the <<actor_somds_provider>>'s MDIB.
```

|<<actor_somds_provider>>
|Listens for <<vol2_clause_dev_30_message_getmdib, {var_label_dev_30_message_getmdib}>> messages, which it responds to with <<vol2_clause_dev_30_message_getmdibresponse, {var_label_dev_30_message_getmdibresponse}>> messages respectively.

|===

==== Referenced Standards

* <<ieee_11073_10207_2017>> Section 7.3.2 Get Service

==== Messages

.Message Interaction Diagram {var_transaction_id}
[plantuml#vol2_figure_dev_30_sequence, target=puml-dev-30-sequence, format=svg, reftext='{figure-caption} {counter:refnum}',width=100%]
....
include::../../plantuml/vol2-figure-dev-30-sequence.puml[]
....

// ---------- GetMdib ---------

[#vol2_clause_dev_30_message_getmdib]
===== {var_label_dev_30_message_getmdib} Message

> The {var_label_dev_30_message_getmdib} message is sent as part of the BICEPS _Get Service_ protocol for allowing <<actor_somds_consumer>>s to retrieve the description and state of an MDIB from a <<actor_somds_provider>>.

Reviewer: ```suggestion
The {var_label_dev_30_message_getmdib} message is sent as part of the BICEPS GET SERVICE for allowing <<actor_somds_consumer>>s to retrieve the description and state of an MDIB from a <<actor_somds_provider>>.
```

[#vol2_clause_dev_30_message_getmdib_trigger_events]
====== Trigger Events

The {var_label_dev_30_message_getmdib} message is sent

1. whenever a <<actor_somds_consumer>> wants to retrieve the description and state of an MDIB of a <<actor_somds_provider>>.

[#vol2_clause_dev_30_message_getmdib_semantics]
====== Message Semantics

The {var_label_dev_30_message_getmdib} message does not contain any further semantics.

[#vol2_clause_dev_30_message_getmdib_expected_actions]
====== Expected Actions

> When a <<actor_somds_consumer>> sends this message, every receiving <<actor_somds_provider>> responds with a <<vol2_clause_dev_30_message_getmdibresponse, {var_label_dev_30_message_getmdibresponse}>> message.

Reviewer: ```suggestion
When a <<actor_somds_consumer>> sends this message, the receiving <<actor_somds_provider>> responds with a <<vol2_clause_dev_30_message_getmdibresponse, {var_label_dev_30_message_getmdibresponse}>> message.
```

// ---------- Get Mdib Response ---------

[#vol2_clause_dev_30_message_getmdibresponse]
===== {var_label_dev_30_message_getmdibresponse} Message

> The {var_label_dev_30_message_getmdibresponse} message is sent as part of the BICEPS _Get Service_ protocol in response to an incoming <<vol2_clause_dev_30_message_getmdib, {var_label_dev_30_message_getmdib}>> message.

Reviewer: ```suggestion
The {var_label_dev_30_message_getmdibresponse} message is sent as part of the BICEPS GET SERVICE in response to an incoming <<vol2_clause_dev_30_message_getmdib, {var_label_dev_30_message_getmdib}>> message.

```
